File: features/customauth/impl/src/main/kotlin/io/element/android/features/customauth/impl/data/DropdownData.kt
/*
 * Copyright 2024 New Vector Ltd.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */

package io.element.android.features.customauth.impl.data

import io.element.android.features.customauth.impl.R

data class DropdownOption(
    val label: String,
    val value: String,
    val iconResId: Int? = null, // Optional drawable resource ID
)

data class CountryCode(
    val code: String,
    val name: String,
    val dialCode: String,
    val flag: String,
)

object DropdownData {
    val PROFESSIONAL_TITLES =
        listOf(
            DropdownOption("MD", "MD"),
            DropdownOption("DO", "DO"),
            DropdownOption("PA-C", "PA-C"),
            DropdownOption("NP", "NP"),
            DropdownOption("RN", "RN"),
            DropdownOption("PharmD", "PharmD"),
            DropdownOption("DDS", "DDS"),
            DropdownOption("DMD", "DMD"),
            DropdownOption("DPT", "DPT"),
            DropdownOption("OTR", "OTR"),
            DropdownOption("SLP", "SLP"),
            DropdownOption("Student", "Student"),
        )

    val COUNTRIES =
        listOf(
            DropdownOption("🇺🇸 United States", "USA"),
            DropdownOption("🇨🇦 Canada", "Canada"),
        )

    val US_STATES =
        listOf(
            // US States
            DropdownOption("Alabama", "AL", R.drawable.al),
            DropdownOption("Alaska", "AK", R.drawable.ak),
            DropdownOption("Arizona", "AZ", R.drawable.az),
            DropdownOption("Arkansas", "AR", R.drawable.ar),
            DropdownOption("California", "CA", R.drawable.ca),
            DropdownOption("Colorado", "CO", R.drawable.co),
            DropdownOption("Connecticut", "CT", R.drawable.ct),
            DropdownOption("Delaware", "DE", R.drawable.de),
            DropdownOption("Florida", "FL", R.drawable.fl),
            DropdownOption("Georgia", "GA", R.drawable.ga),
            DropdownOption("Hawaii", "HI", R.drawable.hi),
            DropdownOption("Idaho", "ID", R.drawable.id),
            DropdownOption("Illinois", "IL", R.drawable.il),
            DropdownOption("Indiana", "IN", R.drawable.`in`),
            DropdownOption("Iowa", "IA", R.drawable.ia),
            DropdownOption("Kansas", "KS", R.drawable.ks),
            DropdownOption("Kentucky", "KY", R.drawable.ky),
            DropdownOption("Louisiana", "LA", R.drawable.la),
            DropdownOption("Maine", "ME", R.drawable.me),
            DropdownOption("Maryland", "MD", R.drawable.md),
            DropdownOption("Massachusetts", "MA", R.drawable.ma),
            DropdownOption("Michigan", "MI", R.drawable.mi),
            DropdownOption("Minnesota", "MN", R.drawable.mn),
            DropdownOption("Mississippi", "MS", R.drawable.ms),
            DropdownOption("Missouri", "MO", R.drawable.mo),
            DropdownOption("Montana", "MT", R.drawable.mt),
            DropdownOption("Nebraska", "NE", R.drawable.ne),
            DropdownOption("Nevada", "NV", R.drawable.nv),
            DropdownOption("New Hampshire", "NH", R.drawable.nh),
            DropdownOption("New Jersey", "NJ", R.drawable.nj),
            DropdownOption("New Mexico", "NM", R.drawable.nm),
            DropdownOption("New York", "NY", R.drawable.ny),
            DropdownOption("North Carolina", "NC", R.drawable.nc),
            DropdownOption("North Dakota", "ND", R.drawable.nd),
            DropdownOption("Ohio", "OH", R.drawable.oh),
            DropdownOption("Oklahoma", "OK", R.drawable.ok),
            DropdownOption("Oregon", "OR", R.drawable.or),
            DropdownOption("Pennsylvania", "PA", R.drawable.pa),
            DropdownOption("Rhode Island", "RI", R.drawable.ri),
            DropdownOption("South Carolina", "SC", R.drawable.sc),
            DropdownOption("South Dakota", "SD", R.drawable.sd),
            DropdownOption("Tennessee", "TN", R.drawable.tn),
            DropdownOption("Texas", "TX", R.drawable.tx),
            DropdownOption("Utah", "UT", R.drawable.ut),
            DropdownOption("Vermont", "VT", R.drawable.vt),
            DropdownOption("Virginia", "VA", R.drawable.va),
            DropdownOption("Washington", "WA", R.drawable.wa),
            DropdownOption("West Virginia", "WV", R.drawable.wv),
            DropdownOption("Wisconsin", "WI", R.drawable.wi),
            DropdownOption("Wyoming", "WY", R.drawable.wy),
            // US Territories (keeping emojis for now since we don't have custom flags)
            DropdownOption("🏴󠁵󠁳󠁤󠁣󠁿 District of Columbia", "DC"),
            DropdownOption("🇵🇷 Puerto Rico", "PR"),
            DropdownOption("🇻🇮 U.S. Virgin Islands", "VI"),
            DropdownOption("🇬🇺 Guam", "GU"),
            DropdownOption("🇦🇸 American Samoa", "AS"),
            DropdownOption("🇲🇵 Northern Mariana Islands", "MP"),
        )

    val CANADIAN_PROVINCES =
        listOf(
            DropdownOption("🏴󠁣󠁡󠁡󠁢󠁿 Alberta", "AB"),
            DropdownOption("🏴󠁣󠁡󠁢󠁣󠁿 British Columbia", "BC"),
            DropdownOption("🏴󠁣󠁡󠁭󠁢󠁿 Manitoba", "MB"),
            DropdownOption("🏴󠁣󠁡󠁮󠁢󠁿 New Brunswick", "NB"),
            DropdownOption("🏴󠁣󠁡󠁮󠁬󠁿 Newfoundland and Labrador", "NL"),
            DropdownOption("🏴󠁣󠁡󠁮󠁴󠁿 Northwest Territories", "NT"),
            DropdownOption("🏴󠁣󠁡󠁮󠁳󠁿 Nova Scotia", "NS"),
            DropdownOption("🏴󠁣󠁡󠁮󠁵󠁿 Nunavut", "NU"),
            DropdownOption("🏴󠁣󠁡󠁯󠁮󠁿 Ontario", "ON"),
            DropdownOption("🏴󠁣󠁡󠁰󠁥󠁿 Prince Edward Island", "PE"),
            DropdownOption("🏴󠁣󠁡󠁱󠁣󠁿 Quebec", "QC"),
            DropdownOption("🏴󠁣󠁡󠁳󠁫󠁿 Saskatchewan", "SK"),
            DropdownOption("🏴󠁣󠁡󠁹󠁴󠁿 Yukon", "YT"),
        )

    val MEDICAL_SPECIALTIES =
        listOf(
            DropdownOption("Student", "Student"),
            DropdownOption("Other", "Other"),
            DropdownOption("Addiction Medicine", "Addiction Medicine"),
            DropdownOption("Allergy and Immunology", "Allergy and Immunology"),
            DropdownOption("Anesthesiology", "Anesthesiology"),
            DropdownOption("Cardiology", "Cardiology"),
            DropdownOption("Cardiothoracic Surgery", "Cardiothoracic Surgery"),
            DropdownOption("Critical Care Medicine", "Critical Care Medicine"),
            DropdownOption("Dermatology", "Dermatology"),
            DropdownOption("Emergency Medicine", "Emergency Medicine"),
            DropdownOption("Endocrinology", "Endocrinology"),
            DropdownOption("Family Medicine", "Family Medicine"),
            DropdownOption("Gastroenterology", "Gastroenterology"),
            DropdownOption("General Surgery", "General Surgery"),
            DropdownOption("Geriatric Medicine", "Geriatric Medicine"),
            DropdownOption("Hematology", "Hematology"),
            DropdownOption("Hematology/Oncology", "Heme/Onc"),
            DropdownOption("Hospitalist", "Hospitalist"),
            DropdownOption("Infectious Disease", "Infectious Disease"),
            DropdownOption("Internal Medicine", "Internal Medicine"),
            DropdownOption("Interventional Cardiology", "Interventional Cardiology"),
            DropdownOption("Interventional Radiology", "Interventional Radiology"),
            DropdownOption("Nephrology", "Nephrology"),
            DropdownOption("Neurology", "Neurology"),
            DropdownOption("Neurosurgery", "Neurosurgery"),
            DropdownOption("Nuclear Medicine", "Nuclear Medicine"),
            DropdownOption("Obesity Medicine", "Obesity Medicine"),
            DropdownOption("Obstetrics & Gynecology", "Obstetrics & Gynecology"),
            DropdownOption("Occupational Medicine", "Occupational Medicine"),
            DropdownOption("Oncology", "Oncology"),
            DropdownOption("Ophthalmology", "Ophthalmology"),
            DropdownOption("Orthopedic Surgery", "Orthopedic Surgery"),
            DropdownOption("Otolaryngology", "Otolaryngology"),
            DropdownOption("Pain Medicine", "Pain Medicine"),
            DropdownOption("Palliative Care", "Palliative Care"),
            DropdownOption("Pathology", "Pathology"),
            DropdownOption("Pediatrics", "Pediatrics"),
            DropdownOption("Physical Medicine & Rehabilitation", "Physical Medicine & Rehabilitation"),
            DropdownOption("Plastic Surgery", "Plastic Surgery"),
            DropdownOption("Podiatry", "Podiatry"),
            DropdownOption("Preventive Medicine", "Preventive Medicine"),
            DropdownOption("Psychiatry", "Psychiatry"),
            DropdownOption("Psychology", "Psychology"),
            DropdownOption("Pulmonology", "Pulmonology"),
            DropdownOption("Radiation Oncology", "Radiation Oncology"),
            DropdownOption("Radiology", "Radiology"),
            DropdownOption("Registered Nurse", "Registered Nurse"),
            DropdownOption("Rheumatology", "Rheumatology"),
            DropdownOption("Sleep Medicine", "Sleep Medicine"),
            DropdownOption("Sports Medicine", "Sports Medicine"),
            DropdownOption("Trauma Surgery", "Trauma Surgery"),
            DropdownOption("Urgent Care", "Urgent Care"),
            DropdownOption("Urology", "Urology"),
            DropdownOption("Vascular Surgery", "Vascular Surgery"),
        )

    val COUNTRY_CODES =
        listOf(
            CountryCode("US", "United States", "+1", "🇺🇸"),
            CountryCode("CA", "Canada", "+1", "🇨🇦"),
        )

    // Helper function to get states/provinces based on country
    fun getStatesForCountry(countryValue: String): List<DropdownOption> {
        return when (countryValue) {
            "USA" -> US_STATES
            "Canada" -> CANADIAN_PROVINCES
            else -> emptyList()
        }
    }

    // Helper function to check if a country requires state selection
    fun countryRequiresState(countryValue: String): Boolean {
        return listOf("USA", "Canada").contains(countryValue)
    }
} 

File: dropdown-package/data/dropdownData.ts
// Dropdown data for registration forms
// You can customize these lists based on your needs

export interface DropdownOption {
  label: string;
  value: string;
}

export const PROFESSIONAL_TITLES: DropdownOption[] = [
  { label: 'MD', value: 'MD' },
  { label: 'DO', value: 'DO' },
  { label: 'PA-C', value: 'PA-C' },
  { label: 'NP', value: 'NP' },
  { label: 'RN', value: 'RN' },
  { label: 'LPN', value: 'LPN' },
  { label: 'CNA', value: 'CNA' },
  { label: 'PharmD', value: 'PharmD' },
  { label: 'DDS', value: 'DDS' },
  { label: 'DMD', value: 'DMD' },
  { label: 'DPT', value: 'DPT' },
  { label: 'OTR', value: 'OTR' },
  { label: 'SLP', value: 'SLP' },
  { label: 'Student', value: 'Student' },
  { label: 'Other', value: 'Other' },
];

export const COUNTRIES: DropdownOption[] = [
  { label: '🇺🇸 United States', value: 'USA' },
  { label: '🇨🇦 Canada', value: 'Canada' },
  { label: '🇲🇽 Mexico', value: 'Mexico' },
  { label: '🇬🇧 United Kingdom', value: 'UK' },
  { label: '🇦🇺 Australia', value: 'Australia' },
  { label: '🇩🇪 Germany', value: 'Germany' },
  { label: '🇫🇷 France', value: 'France' },
  { label: '🇮🇹 Italy', value: 'Italy' },
  { label: '🇪🇸 Spain', value: 'Spain' },
  { label: '🇳🇱 Netherlands', value: 'Netherlands' },
  { label: '🇧🇷 Brazil', value: 'Brazil' },
  { label: '🇦🇷 Argentina', value: 'Argentina' },
  { label: '🇯🇵 Japan', value: 'Japan' },
  { label: '🇰🇷 South Korea', value: 'South Korea' },
  { label: '🇮🇳 India', value: 'India' },
  { label: '🇨🇳 China', value: 'China' },
];

export const US_STATES: DropdownOption[] = [
  { label: 'Alabama', value: 'AL' },
  { label: 'Alaska', value: 'AK' },
  { label: 'Arizona', value: 'AZ' },
  { label: 'Arkansas', value: 'AR' },
  { label: 'California', value: 'CA' },
  { label: 'Colorado', value: 'CO' },
  { label: 'Connecticut', value: 'CT' },
  { label: 'Delaware', value: 'DE' },
  { label: 'Florida', value: 'FL' },
  { label: 'Georgia', value: 'GA' },
  { label: 'Hawaii', value: 'HI' },
  { label: 'Idaho', value: 'ID' },
  { label: 'Illinois', value: 'IL' },
  { label: 'Indiana', value: 'IN' },
  { label: 'Iowa', value: 'IA' },
  { label: 'Kansas', value: 'KS' },
  { label: 'Kentucky', value: 'KY' },
  { label: 'Louisiana', value: 'LA' },
  { label: 'Maine', value: 'ME' },
  { label: 'Maryland', value: 'MD' },
  { label: 'Massachusetts', value: 'MA' },
  { label: 'Michigan', value: 'MI' },
  { label: 'Minnesota', value: 'MN' },
  { label: 'Mississippi', value: 'MS' },
  { label: 'Missouri', value: 'MO' },
  { label: 'Montana', value: 'MT' },
  { label: 'Nebraska', value: 'NE' },
  { label: 'Nevada', value: 'NV' },
  { label: 'New Hampshire', value: 'NH' },
  { label: 'New Jersey', value: 'NJ' },
  { label: 'New Mexico', value: 'NM' },
  { label: 'New York', value: 'NY' },
  { label: 'North Carolina', value: 'NC' },
  { label: 'North Dakota', value: 'ND' },
  { label: 'Ohio', value: 'OH' },
  { label: 'Oklahoma', value: 'OK' },
  { label: 'Oregon', value: 'OR' },
  { label: 'Pennsylvania', value: 'PA' },
  { label: 'Rhode Island', value: 'RI' },
  { label: 'South Carolina', value: 'SC' },
  { label: 'South Dakota', value: 'SD' },
  { label: 'Tennessee', value: 'TN' },
  { label: 'Texas', value: 'TX' },
  { label: 'Utah', value: 'UT' },
  { label: 'Vermont', value: 'VT' },
  { label: 'Virginia', value: 'VA' },
  { label: 'Washington', value: 'WA' },
  { label: 'West Virginia', value: 'WV' },
  { label: 'Wisconsin', value: 'WI' },
  { label: 'Wyoming', value: 'WY' },
];

export const CANADIAN_PROVINCES: DropdownOption[] = [
  { label: 'Alberta', value: 'AB' },
  { label: 'British Columbia', value: 'BC' },
  { label: 'Manitoba', value: 'MB' },
  { label: 'New Brunswick', value: 'NB' },
  { label: 'Newfoundland and Labrador', value: 'NL' },
  { label: 'Northwest Territories', value: 'NT' },
  { label: 'Nova Scotia', value: 'NS' },
  { label: 'Nunavut', value: 'NU' },
  { label: 'Ontario', value: 'ON' },
  { label: 'Prince Edward Island', value: 'PE' },
  { label: 'Quebec', value: 'QC' },
  { label: 'Saskatchewan', value: 'SK' },
  { label: 'Yukon', value: 'YT' },
];

export const MEDICAL_SPECIALTIES: DropdownOption[] = [
  { label: 'Student', value: 'Student' },
  { label: 'Other', value: 'Other' },
  { label: 'Addiction Medicine', value: 'Addiction Medicine' },
  { label: 'Allergy and Immunology', value: 'Allergy and Immunology' },
  { label: 'Anesthesiology', value: 'Anesthesiology' },
  { label: 'Cardiology', value: 'Cardiology' },
  { label: 'Cardiothoracic Surgery', value: 'Cardiothoracic Surgery' },
  { label: 'Critical Care Medicine', value: 'Critical Care Medicine' },
  { label: 'Dermatology', value: 'Dermatology' },
  { label: 'Emergency Medicine', value: 'Emergency Medicine' },
  { label: 'Endocrinology', value: 'Endocrinology' },
  { label: 'Family Medicine', value: 'Family Medicine' },
  { label: 'Gastroenterology', value: 'Gastroenterology' },
  { label: 'General Surgery', value: 'General Surgery' },
  { label: 'Geriatric Medicine', value: 'Geriatric Medicine' },
  { label: 'Hematology', value: 'Hematology' },
  { label: 'Hematology/Oncology', value: 'Heme/Onc' },
  { label: 'Hospitalist', value: 'Hospitalist' },
  { label: 'Infectious Disease', value: 'Infectious Disease' },
  { label: 'Internal Medicine', value: 'Internal Medicine' },
  { label: 'Interventional Cardiology', value: 'Interventional Cardiology' },
  { label: 'Interventional Radiology', value: 'Interventional Radiology' },
  { label: 'Nephrology', value: 'Nephrology' },
  { label: 'Neurology', value: 'Neurology' },
  { label: 'Neurosurgery', value: 'Neurosurgery' },
  { label: 'Nuclear Medicine', value: 'Nuclear Medicine' },
  { label: 'Obesity Medicine', value: 'Obesity Medicine' },
  { label: 'Obstetrics & Gynecology', value: 'Obstetrics & Gynecology' },
  { label: 'Occupational Medicine', value: 'Occupational Medicine' },
  { label: 'Oncology', value: 'Oncology' },
  { label: 'Ophthalmology', value: 'Ophthalmology' },
  { label: 'Orthopedic Surgery', value: 'Orthopedic Surgery' },
  { label: 'Otolaryngology', value: 'Otolaryngology' },
  { label: 'Pain Medicine', value: 'Pain Medicine' },
  { label: 'Palliative Care', value: 'Palliative Care' },
  { label: 'Pathology', value: 'Pathology' },
  { label: 'Pediatrics', value: 'Pediatrics' },
  { label: 'Physical Medicine & Rehabilitation', value: 'Physical Medicine & Rehabilitation' },
  { label: 'Plastic Surgery', value: 'Plastic Surgery' },
  { label: 'Podiatry', value: 'Podiatry' },
  { label: 'Preventive Medicine', value: 'Preventive Medicine' },
  { label: 'Psychiatry', value: 'Psychiatry' },
  { label: 'Psychology', value: 'Psychology' },
  { label: 'Pulmonology', value: 'Pulmonology' },
  { label: 'Radiation Oncology', value: 'Radiation Oncology' },
  { label: 'Radiology', value: 'Radiology' },
  { label: 'Registered Nurse', value: 'Registered Nurse' },
  { label: 'Rheumatology', value: 'Rheumatology' },
  { label: 'Sleep Medicine', value: 'Sleep Medicine' },
  { label: 'Sports Medicine', value: 'Sports Medicine' },
  { label: 'Trauma Surgery', value: 'Trauma Surgery' },
  { label: 'Urgent Care', value: 'Urgent Care' },
  { label: 'Urology', value: 'Urology' },
  { label: 'Vascular Surgery', value: 'Vascular Surgery' },
];

// Country codes for phone number input
export interface CountryCode {
  code: string;
  name: string;
  dialCode: string;
  flag: string;
}

export const COUNTRY_CODES: CountryCode[] = [
  { code: 'US', name: 'United States', dialCode: '+1', flag: '🇺🇸' },
  { code: 'CA', name: 'Canada', dialCode: '+1', flag: '🇨🇦' },
  { code: 'MX', name: 'Mexico', dialCode: '+52', flag: '🇲🇽' },
  { code: 'GB', name: 'United Kingdom', dialCode: '+44', flag: '🇬🇧' },
  { code: 'AU', name: 'Australia', dialCode: '+61', flag: '🇦🇺' },
  { code: 'DE', name: 'Germany', dialCode: '+49', flag: '🇩🇪' },
  { code: 'FR', name: 'France', dialCode: '+33', flag: '🇫🇷' },
  { code: 'IT', name: 'Italy', dialCode: '+39', flag: '🇮🇹' },
  { code: 'ES', name: 'Spain', dialCode: '+34', flag: '🇪🇸' },
  { code: 'NL', name: 'Netherlands', dialCode: '+31', flag: '🇳🇱' },
  { code: 'BR', name: 'Brazil', dialCode: '+55', flag: '🇧🇷' },
  { code: 'AR', name: 'Argentina', dialCode: '+54', flag: '🇦🇷' },
  { code: 'JP', name: 'Japan', dialCode: '+81', flag: '🇯🇵' },
  { code: 'KR', name: 'South Korea', dialCode: '+82', flag: '🇰🇷' },
  { code: 'IN', name: 'India', dialCode: '+91', flag: '🇮🇳' },
  { code: 'CN', name: 'China', dialCode: '+86', flag: '🇨🇳' },
];

// Helper function to get states/provinces based on country
export const getStatesForCountry = (countryValue: string): DropdownOption[] => {
  switch (countryValue) {
    case 'USA':
      return US_STATES;
    case 'Canada':
      return CANADIAN_PROVINCES;
    default:
      return [];
  }
};

// Helper function to check if a country requires state selection
export const countryRequiresState = (countryValue: string): boolean => {
  return ['USA', 'Canada'].includes(countryValue);
}; 